### PR TITLE
Audit Oauth2 Access Token Requests

### PR DESCRIPTION
**Pull request by dima767**

### CODE REVIEW

File: support/cas-server-support-oauth/src/main/java/org/apereo/cas/support/oauth/web/response/accesstoken/ext/BaseAccessTokenGrantRequestExtractor.java [hunk 1]
```diff
@@ -50,6 +51,9 @@
      * @param response the response
      * @return the access token request data holder
      */
+    @Audit(action = "OAUTH2_ACCESS_TOKEN_REQUEST",
+        actionResolverName = "OAUTH2_ACCESS_TOKEN_REQUEST_ACTION_RESOLVER",
+        resourceResolverName = "OAUTH2_ACCESS_TOKEN_REQUEST_RESOURCE_RESOLVER")
```
Comment (mmoayyed) [inline support/cas-server-support-oauth/src/main/java/org/apereo/cas/support/oauth/web/response/accesstoken/ext/BaseAccessTokenGrantRequestExtractor.java]: Please document these.

File: support/cas-server-support-oauth/src/main/java/org/apereo/cas/support/oauth/web/audit/AccessTokenGrantRequestAuditResourceResolver.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    public String[] resolveFrom(final JoinPoint auditableTarget, final Object retval) {
+        Objects.requireNonNull(retval, "AccessTokenRequestDataHolder must not be null");
+        final AccessTokenRequestDataHolder accessTokenRequest = AccessTokenRequestDataHolder.class.cast(retval);
+
+        final String result = new ToStringBuilder(this, NO_CLASS_NAME_STYLE)
+                .append("oauth_token", accessTokenRequest.getToken().getId())
+                .append("client_id", accessTokenRequest.getRegisteredService().getClientId())
+                .append("client_service", accessTokenRequest.getService().getId())
+                .append("grant_type", accessTokenRequest.getGrantType().getType())
+                .append("scopes", accessTokenRequest.getScopes())
```
Comment (mmoayyed) [inline support/cas-server-support-oauth/src/main/java/org/apereo/cas/support/oauth/web/audit/AccessTokenGrantRequestAuditResourceResolver.java]: Is there a possibility that we might run into an NPE here? or that the values be null?

File: support/cas-server-support-oauth/src/main/java/org/apereo/cas/support/oauth/web/response/accesstoken/ext/BaseAccessTokenGrantRequestExtractor.java (hunk 1 above)
Comment (dima767) [reply support/cas-server-support-oauth/src/main/java/org/apereo/cas/support/oauth/web/response/accesstoken/ext/BaseAccessTokenGrantRequestExtractor.java]: Will do

File: support/cas-server-support-oauth/src/main/java/org/apereo/cas/support/oauth/web/audit/AccessTokenGrantRequestAuditResourceResolver.java (hunk 2 above)
Comment (dima767) [reply support/cas-server-support-oauth/src/main/java/org/apereo/cas/support/oauth/web/audit/AccessTokenGrantRequestAuditResourceResolver.java]: Will check

Comment (dima767) [reply support/cas-server-support-oauth/src/main/java/org/apereo/cas/support/oauth/web/audit/AccessTokenGrantRequestAuditResourceResolver.java]: OK, traced through the code. According to current impl. only the token could be null. Pusheed the change.

**Pull request merged**